components/MnemonicView.tsx

```diff
@@ -121,11 +131,7 @@ const MnemonicView: FC<MnemonicViewProps> = ({
   }
 
   useEffect(() => {
-    $setCurrentWords(isReadOnly ? value : new Array(12).fill(''))
-  }, [])
-
-  useEffect(() => {
-    $setCurrentWords(value.length ? value : new Array(12).fill(''))
+    $setCurrentWords(getCurrentWords(value, wordsAmount))
```

Comment:
Hey, I pulled this down to see if we could avoid having to keep a copy of the words in state and update via useEffect. IMO, what you have is fine, but since we can derive the state that we want (an array of strings with length equal to wordsAmount), we may as well.

Feel free to merge my PR into yours or close it if you don't agree. [PR is right here](https://github.com/iron-fish/ui-kit/pull/113)

Reply:
Good point, thanks. I will use your suggestion.